cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationFinder.java

```diff
@@ -88,6 +89,37 @@ public static Properties readConfigrationFile(Path cobigenConfigFile) {
     return props;
   }
 
+  /**
+   * checks if .cobigen properties file contains any add-generated-annotation
+   *
+   * @return the value of add-generated-annotation
+   *
+   * @throws FileNotFoundException
+   */
+  public static boolean checkGeneratedAnnotationInProperties() throws FileNotFoundException {
+
+    Boolean defaultGeneratedAnnotation = true;
+    Path cobigenHome = CobiGenPaths.getCobiGenHomePath();
+    Path propertiesPath = cobigenHome.resolve(ConfigurationConstants.COBIGEN_CONFIG_FILE);
+
+    // check if .cobigen properties file has add-generated annotation set to true
+    if (Files.exists(propertiesPath)) {
+
+      Properties props = ConfigurationFinder.readConfigurationFile(propertiesPath);
+
+      if (props.containsKey(ConfigurationConstants.ADD_GENERATED_ANNOTATION))
+        defaultGeneratedAnnotation = Boolean.valueOf(ConfigurationConstants.ADD_GENERATED_ANNOTATION);
+      else {
+        // add the default value in .cobigen properties
+        FileSystemUtil.writeOutputToFile(propertiesPath, ConfigurationConstants.ADD_GENERATED_ANNOTATION + "=true");
+      }
+    } else {
+      // add the default value in .cobigen properties
+      FileSystemUtil.writeOutputToFile(propertiesPath, ConfigurationConstants.ADD_GENERATED_ANNOTATION + "=true");
```

Comment:
Please replace the writeOutputToFile with java.util.Properties reader/writer.
Take a look at: https://www.baeldung.com/java-properties